packages/legend-studio/src/stores/editor-state/GraphGenerationState.ts

```diff
@@ -128,14 +133,31 @@ export class GraphGenerationState {
   get supportedFileGenerationConfigurationsForCurrentElement(): GenerationConfigurationDescription[] {
     if (this.editorStore.currentEditorState instanceof ElementEditorState) {
       const currentElement = this.editorStore.currentEditorState.element;
-      if (
-        currentElement instanceof Class ||
-        currentElement instanceof Enumeration
-      ) {
-        return this.fileGenerationConfigurations
-          .slice()
-          .sort((a, b): number => a.label.localeCompare(b.label));
-      }
+      const getExtraFileGenerationScopeFilters = this.editorStore.pluginManager
+        .getStudioPlugins()
+        .flatMap(
+          (plugin) =>
+            (
+              plugin as DSL_LegendStudioPlugin_Extension
+            ).getExtraFileGenerationScopeFilters?.() ?? [],
+        )
+        .concat([
```

Comment:
Could we add a comment here to say right now we only allow `classes` and `enumerations` for all types of generations? Also, we should remove `ConcreteFunctionDefinition`. We never support that for other generation types